dev/flake.nix

```diff
@@ -4,6 +4,7 @@
     flake-parts.url = "github:hercules-ci/flake-parts";
     flake-root.url = "github:srid/flake-root";
     treefmt-nix.url = "github:numtide/treefmt-nix";
+    services-flake.url = "path:../.";
```

Comment:
You can also do `services-flake = {}`. This will force the use of `--overridde-input`, which is what we want.

Reply:
In fact, we should do this for test/ as well. All but example I'd say.

https://github.com/juspay/services-flake/blob/29781d10d08ab704dacf58cad35cddac7d19d5bd/test/flake.nix#L7

Reply:
> You can also do services-flake = {}

That makes sense!

I had changed it to the github flake url because devShell would throw this error otherwise:
<img width="1336" alt="Screenshot 2024-02-20 at 1 49 38 AM" src="https://github.com/juspay/services-flake/assets/23645788/45910820-d97a-4978-b610-e52197af272f">

Reply:
I guess we cannot do that

Reply:
if i go in `nix develop` without `direnv`, i don't get the error.

```
abhishek in 🌐 abhishek-2M352918QDA9 in services-flake on  fix/ci-treefmt [+] via ❄️  impure (nix-shell-env)
❯ direnv deny
direnv: error /home/abhishek/dev/nix/services-flake/.envrc is blocked. Run `direnv allow` to approve its content

abhishek in 🌐 abhishek-2M352918QDA9 in services-flake on  fix/ci-treefmt [+]
❯ nix develop ./dev --override-input services-flake .

warning: Git tree '/home/abhishek/dev/nix/services-flake' is dirty
warning: Git tree '/home/abhishek/dev/nix/services-flake' is dirty
warning: not writing modified lock file of flake 'git+file:///home/abhishek/dev/nix/services-flake?dir=dev':
• Updated input 'services-flake':
    'path:../.?lastModified=0&narHash=sha256-X97SAw/4FVSb%2B4SORDS5Kzuvqs5DoD7oS4CQ3BjMXW4%3D' (1970-01-01)
  → 'git+file:///home/abhishek/dev/nix/services-flake' (2024-02-19)

🍎🍎 Run 'just <recipe>' to get started
Available recipes:
    default
    ex           # Run example
    fmt          # Auto-format the project tree
    test service # Run native test for a specific service
    test-all     # Run native tests for all the services

abhishek in 🌐 abhishek-2M352918QDA9 in services-flake on  fix/ci-treefmt [+] via ❄️  impure (nix-shell-env)
❯
```

Reply:
I don’t think adding `—override-input services-flake .` to enter devShell is a good default.

I think it is simpler to keep `services-flake.url = “github:juspay/services-falke”` for the dev flake at least.

Reply:
Maybe also add a comment specifying that it will be overriden in CI, to check the formatting on the latest source.

Reply:
@srid any thoughts? I was think I kinda of merging otherwise

Reply:
Sure, go ahead.

FWIW, I just specify --override-input in `.envrc`, but if you want to support plain `nix develop` it makes sense to keep the full flake input.

https://github.com/srid/haskell-flake/blob/master/.envrc

Reply:
I wonder how is `treefmt` check on projectRoot working in haskell-flake, from what I see it is set to `../.`: https://github.com/srid/haskell-flake/blob/5113f700d6e92199fbe0574f7d12c775bb169702/dev/flake.nix#L19

Reply:
And[ I don’t see haskell-flake input in dev flake](https://github.com/srid/haskell-flake/blob/5113f700d6e92199fbe0574f7d12c775bb169702/dev/flake.nix#L3-L7), what does `.envrc` override?

Reply:
I think that explains what you were facing here: https://nixos.zulipchat.com/#narrow/stream/413948-nixos/topic/Self-hosted.20GitHub.20runners/near/421222981

Reply:
It was an oversight; fixed here https://github.com/srid/haskell-flake/pull/233